tests/test_agent_manager.py

```diff
@@ -744,7 +744,8 @@ async def test_session_creation_fails(server, environment, async_finalizer, capl
     assert len(session_manager._sessions) == 0
 
     # Remove connectivity to the database
-    await data.disconnect()
+    data.ENGINE = None
```

Comment:
is there a reason why we prefer to do this instead of `data.ENGINE.dispose()`?

Reply:
Yes, `dispose()` won't kill connectivity (see ` This disposes of the engine’s underlying connection pool and replaces it with a new one that’s empty` in https://docs.sqlalchemy.org/en/20/core/connections.html#engine-disposal).

I'll still replace this with a call to `stop_engine`